url-updater/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -250,37 +265,30 @@ private boolean checkDownloadUrl(String url, UrlVersion urlVersion, OperatingSys
    * @return {@code true} if the download was checked successfully, {@code false} otherwise.
    */
   private boolean checkDownloadUrl(String url, UrlVersion urlVersion, OperatingSystem os,
-      SystemArchitecture architecture) {
+      SystemArchitecture architecture, String checksum) {
 
-    HttpResponse<InputStream> response = doGetResponseAsStream(url);
+    HttpResponse<?> response = doCheckDownloadViaHeadRequest(url);
```

Comment:
It would be even smarter to distinguish if we already have a checksum or not:
* if we already have a checksum given, a HEAD request is sufficient
* otherwise we need the full download and can do only a GET - currently you always do HEAD and then additionally to a GET request what is overhead.

Reply:
We will do this after the merge to not block the merge.